cmd/copy/main.go

```diff
@@ -50,12 +50,40 @@ func copy(source checks.DefaultChecks, destination checks.DefaultChecks) error {
 			return err
 		}
 		for _, snapshot := range snapshots {
-			err := destination.CopySnapshots(snapshot, instance.Destination)
-			if err != nil {
-				log.WithFields(log.Fields{
-					"Snapshot": *snapshot.DBSnapshotIdentifier,
-				}).Errorf("Could not copy snapshot: %s", err)
-				return err
+			if *snapshot.SnapshotType == "automated" {
+				err := destination.PostDatadogChecks(snapshot, "rdscheck.status", "ok", "copy")
+				if err != nil {
+					log.WithError(err).Error("Could not update datadog status")
+					return err
+				}
+				var preSignedUrl string
+				cleanArn := destination.CleanArn(snapshot)
+				if *snapshot.Encrypted {
+					preSignedUrl, err = source.PreSignUrl(instance.Destination, *snapshot.DBSnapshotArn, instance.KmsID, cleanArn)
+					if err != nil {
+						log.WithFields(log.Fields{
+							"snapshot": *snapshot.DBSnapshotIdentifier,
+						}).Errorf("Could not presigned the url: %s", err)
```

Comment:
Why not use `WithError` here as well? You can do `WithFields` and `WithError` on the same statement.